src/Functions/keyvaluepair/src/impl/state/State.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstddef>
+
+namespace DB
+{
+
+enum State
+{
+    WAITING_KEY,
+    READING_KEY,
+    READING_ENCLOSED_KEY,
+    READING_KV_DELIMITER,
+    WAITING_VALUE,
+    READING_VALUE,
+    READING_ENCLOSED_VALUE,
+    READING_EMPTY_VALUE,
+    FLUSH_PAIR,
+    END
```

Comment:
Let's add doc comment with explanation for each state. E.g. what is `enclosed_key/value`, `flush`, the difference between waiting and reading?